python/ray/autoscaler/autoscaler.py

```diff
@@ -133,6 +138,11 @@ def update(self):
                                 "Too many errors, abort.")
                 raise e
 
+    def _node_resources(self, node_id):
+        instance_name = self.node_types[node_id]
```

Comment:
I think this can be changed to `self.provider.node_tags(node_id)[TAG_RAY_INSTANCE_TYPE]`. Or, we can add a helper method to node provider to simplify this.

Reply:
I feel like it makes more sense on this side of the node provider abstraction